server_test.go

```diff
@@ -208,6 +208,51 @@ func TestOpenStatRace(t *testing.T) {
 	checkServerAllocator(t, server)
 }
 
+func TestOpenWithPermissions(t *testing.T) {
+	client, server := clientServerPair(t)
+	defer client.Close()
+	defer server.Close()
+
+	tmppath := path.Join(os.TempDir(), "open_permissions")
+	pflags := flags(os.O_RDWR | os.O_CREATE | os.O_TRUNC)
```

Comment:
You need to not use the `os.O_*` flags. SFTP does not use the same flags.

For SFTP:
```
        #define SSH_FXF_READ            0x00000001
        #define SSH_FXF_WRITE           0x00000002
        #define SSH_FXF_CREAT           0x00000008
        #define SSH_FXF_TRUNC           0x00000010
```

for `os` imported from `syscall`:
```
	O_RDWR                           = 0x2
	O_CREAT                          = 0x40
	O_TRUNC                          = 0x200
```

If the test was working with these flags, then 😰 uh-oh.

Reply:
I am wrong here. `flags` does the conversion. (Excellent name there.)